3.12.0 release is broken: error: no matching member function for call to 'reextent'

```
In file included from /disk-samsung/freebsd-ports/science/qmcpack/work/qmcpack-3.12.0/src/QMCTools/ppconvert/src/common/IOASCII.cc:16:
In file included from /disk-samsung/freebsd-ports/science/qmcpack/work/qmcpack-3.12.0/src/QMCTools/ppconvert/src/common/IO.h:20:
In file included from /disk-samsung/freebsd-ports/science/qmcpack/work/qmcpack-3.12.0/src/QMCTools/ppconvert/src/common/IOBase.h:26:
In file included from /disk-samsung/freebsd-ports/science/qmcpack/work/qmcpack-3.12.0/src/QMCTools/ppconvert/src/common/IOVar.h:20:
In file included from /disk-samsung/freebsd-ports/science/qmcpack/work/qmcpack-3.12.0/src/QMCTools/ppconvert/src/common/IOVarASCII.h:20:
In file included from /disk-samsung/freebsd-ports/science/qmcpack/work/qmcpack-3.12.0/src/QMCTools/ppconvert/src/common/IOVarBase.h:22:
/disk-samsung/freebsd-ports/science/qmcpack/work/qmcpack-3.12.0/src/QMCTools/ppconvert/src/common/Blitz.h:75:54: error: no matching member function for call to 'reextent'
        void resizeAndPreserve(sizes_type sizes){base_type::reextent(sizes);}
                                                 ~~~~~~~~~~~^~~~~~~~
/disk-samsung/freebsd-ports/science/qmcpack/work/qmcpack-3.12.0/src/QMCTools/ppconvert/src/common/IOVarASCII.h:352:14: note: in instantiation of member function 'Array<bool, 4, boost::multi::array<bool, 4>>::resizeAndPreserve' requested here
  ArrayValue.resizeAndPreserve(dims);
             ^
/disk-samsung/freebsd-ports/science/qmcpack/work/qmcpack-3.12.0/src/QMCTools/ppconvert/src/common/IOVarASCII.h:173:3: note: in instantiation of member function 'IO::IOVarASCII<bool, 4>::Resize' requested here
  IOVarASCII(std::string name) { Name = name; }
  ^
/disk-samsung/freebsd-ports/science/qmcpack/work/qmcpack-3.12.0/src/QMCTools/ppconvert/src/common/IOASCII.cc:621:35: note: in instantiation of member function 'IO::IOVarASCII<bool, 4>::IOVarASCII' requested here
        IOVarASCII<bool,4> *newVar = new IOVarASCII<bool,4>(name);
                                         ^
/disk-samsung/freebsd-ports/science/qmcpack/work/qmcpack-3.12.0/external_codes/boost_multi/multi/array.hpp:910:7: note: candidate function not viable: no known conversion from 'Array<bool, 4, boost::multi::array<bool, 4>>::sizes_type' (aka 'std::tuple<long, long, long, long>') to 'const typename array<bool, 4>::extensions_type' (aka 'const extensions_t<4L>') for 1st argument
        auto reextent(typename array::extensions_type const& x) -> array& {
             ^
/disk-samsung/freebsd-ports/science/qmcpack/work/qmcpack-3.12.0/external_codes/boost_multi/multi/array.hpp:941:7: note: candidate function not viable: requires 2 arguments, but 1 was provided
        auto reextent(typename array::extensions_type const& x, typename array::element const& e) -> array& {
             ^
```

clang-13
OS: FreeBSD 13

--- a/src/QMCTools/ppconvert/src/common/Blitz.h
+++ b/src/QMCTools/ppconvert/src/common/Blitz.h
@@ -73,8 +73,9 @@ struct Array : base_type{
 	sizes_type shape() const{return base_type::sizes();}
 	void resize(sizes_type sizes){resizeAndPreserve(sizes);}
 	void resizeAndPreserve(sizes_type sizes){
-		// must do a reinterpret_cast due to failure with libc++ type automatic conversion
-		base_type::reextent(reinterpret_cast<typename base_type::extensions_type const&>(sizes));
+		// explicit conversion due to failure with libc++ type automatic conversion
+		 	base_type::reextent(std::apply([](auto... ss){return typename base_type::extensions_type{static_cast<typename base_type::size_type>(ss)...};}, sizes));
+
 	}
 	template<class... Ints>
 	void resize(Ints... ns){base_type::reextent(std::make_tuple(ns...));}

--- a/src/QMCTools/ppconvert/src/common/Blitz.h
+++ b/src/QMCTools/ppconvert/src/common/Blitz.h
@@ -72,7 +72,10 @@ struct Array : base_type{
 	using sizes_type = decltype(std::declval<base_type const&>().sizes());
 	sizes_type shape() const{return base_type::sizes();}
 	void resize(sizes_type sizes){resizeAndPreserve(sizes);}
-	void resizeAndPreserve(sizes_type sizes){base_type::reextent(sizes);}
+	void resizeAndPreserve(sizes_type sizes){
+		// must do a reinterpret_cast due to failure with libc++ type automatic conversion
+		base_type::reextent(reinterpret_cast<typename base_type::extensions_type const&>(sizes));
+	}
 	template<class... Ints>
 	void resize(Ints... ns){base_type::reextent(std::make_tuple(ns...));}
 	typename base_type::element_ptr       data()      {return base_type::data_elements();}

--- a/src/QMCWaveFunctions/ElectronGas/HEGGrid.h
+++ b/src/QMCWaveFunctions/ElectronGas/HEGGrid.h
@@ -19,6 +19,7 @@
 #include "Lattice/CrystalLattice.h"
 #include <map>
 #include <optional>
+#include <array>
 
 namespace qmcplusplus
 {

